extensions/amp-accordion/1.0/animations.js

```diff
@@ -0,0 +1,154 @@
+      opacity: oldOpacity,
+      overflowY: oldOverflowY,
+    } = content.style;
+
+    // Measure the expanded height. This is relatively heavy with a sync
+    // layout. But no way around it. The hope that the `commitStyles` API
+    // may eventually make this unneeded.
+    content.style.height = '0';
+    content.style.opacity = '0';
+    content.style.overflowY = 'auto';
+    content.hidden = false;
+    const targetHeight = content./*OK*/ scrollHeight;
+
+    // Reset back. The animation will take care of these properties
+    // going forward.
+    content.style.height = oldHeight;
+    content.style.opacity = oldOpacity;
+    content.style.overflowY = oldOverflowY;
+
+    const duration = getTransitionDuration(targetHeight);
+
+    return content.animate(
+      [
+        {
+          height: 0,
+          opacity: 0,
+          overflowY: 'hidden',
+        },
+        {
+          height: targetHeight + 'px',
+          opacity: 1,
+          overflowY: 'hidden',
+        },
+      ],
+      {
+        easing: EXPAND_CURVE,
+        duration,
+      }
+    );
+  });
+}
+
+/**
+ * @param {!Element} content
+ * @return {!UnlistenDef}
+ */
+export function animateCollapse(content) {
+  return animate(
+    content,
+    () => {
+      // Measure the starting height.
+      // This looks ugly, but avoids weird `hidden` state management in the
+      // component itself. Most importantly, this has no performance issues
+      // by itself b/c flipping `hidden` back and force before measure has
+      // no effect - the `useLayoutEffect` assures this.
+      content.hidden = false;
+      const startHeight = content./*OK*/ offsetHeight;
+
+      const duration = getTransitionDuration(startHeight);
+
+      return content.animate(
+        [
+          {
+            height: startHeight + 'px',
+            opacity: 1,
+            overflowY: 'hidden',
+          },
+          {
+            height: '0',
+            opacity: 0,
+            overflowY: 'hidden',
+          },
+        ],
+        {
+          easing: COLLAPSE_CURVE,
+          duration,
+        }
+      );
+    },
+    () => {
+      // Undo the `content.hidden = true` above.
+      content.hidden = true;
+    }
+  );
+}
+
+/**
+ * @param {!Element} element
+ * @param {function():!Animation} prepare
+ * @param {function()=} cleanup
+ * @return {!UnlistenDef}
+ */
+function animate(element, prepare, cleanup = undefined) {
+  let player = prepare();
+  player.onfinish = player.oncancel = () => {
+    player = null;
+    if (cleanup) {
+      cleanup();
+    }
+  };
+  return () => {
+    if (player) {
+      player.cancel();
+    }
+  };
+}
+
+/**
+ * Calculates transition duration from vertical distance traveled
+ * @param {number} dy
+ * @return {number}
+ */
+function getTransitionDuration(dy) {
+  const maxY = window./*OK*/ innerHeight;
+  const distanceAdjustedDuration =
+    (Math.abs(dy) / maxY) * MAX_TRANSITION_DURATION;
+  return Math.min(
```

Comment:
This can be `clamp` for `math.js`

Reply:
That's the thing we recently rolled back :( The use of `devAssert` is still a problem.